sysdeps/managarm/generic/mount.cpp

```diff
@@ -12,13 +12,16 @@
 namespace mlibc {
 
 int sys_mount(const char *source, const char *target,
-		const char *fstype, unsigned long, const void *) {
+		const char *fstype, unsigned long, const void *data) {
 	SignalGuard sguard;
 
 	managarm::posix::MountRequest<MemoryAllocator> req(getSysdepsAllocator());
 	req.set_path(frg::string<MemoryAllocator>(getSysdepsAllocator(), source));
 	req.set_target_path(frg::string<MemoryAllocator>(getSysdepsAllocator(), target));
 	req.set_fs_type(frg::string<MemoryAllocator>(getSysdepsAllocator(), fstype));
+	if(data){
+		req.set_arguments(frg::string<MemoryAllocator>(getSysdepsAllocator(), (char*)data));
```

Comment:
Is the extra data always a string? (Is it just the mount options as specified by `-o` to `mount(8)`?)

Reply:
I'm not aware of any cases in which this data isn't a string, but I don't think it's is required to be a string. This is mainly a temporary measure until I can think of a better way to approach this.

Reply:
`mount(2)` on my system says:

> The  data argument is interpreted by the different filesystems. Typically it is a string of comma-separated options understood by this filesystem.  See mount(8) for details of the options available for each filesystem type.

I think that, practically speaking, we would be fine by just interpreting this as a string in almost any occasion. However, I don't think it's technically correct.

Reply:
That seems fine to me then. Since the man page directs you to `mount(8)` then it'd be difficult to specify binary data (maybe check `mount` source code?)

Reply:
I might check the source later, but it's not very high-priority for me. I was thinking about implementing some sort of fs-specific argument decoding dispatch mechanism, but I'm not sure how I'd go about doing this and it'd probably require the sysdep to be aware of every possible filesystem - which would be impractical at best

Reply:
I'm just throwing around ideas here, but isn't it also possible to have a ``pushMemory`` action similarly to how we have a ``pushDescriptor`` action that pushes an arbitrary piece of memory across process boundaries? Or maybe a mechanism which turns a piece of memory into a descriptor? IIRC that would also help at other points in the managarm sysdeps.